prime-dotnet-webapi/Services/PlrProviderService.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Prime.Models;
+
+namespace Prime.Services
+{
+    public class PlrProviderService : BaseService, IPlrProviderService
+    {
+        private readonly ILogger _logger;
+
+        private readonly List<IdentifierType> _identifierTypes;
+
+        public PlrProviderService(
+            ApiDbContext context,
+            IHttpContextAccessor httpContext,
+            ILogger<PlrProviderService> logger)
+            : base(context, httpContext)
+        {
+            _logger = logger;
+
+            var task = _context.Set<IdentifierType>()
+                            .AsNoTracking()
+                            .ToListAsync();
+            _identifierTypes = task.Result;
+        }
+
+        //        public async Task<int> CreateOrUpdatePlrProviderAsync(PlrProvider dataObject)
+        public int CreateOrUpdatePlrProvider(PlrProvider dataObject, bool expectExists = false)
```

Comment:
Is this method intentionally written to have a synchronous signature? Why is this comment here?